server/src/internalClusterTest/java/org/elasticsearch/search/fetch/subphase/highlight/HighlighterSearchIT.java

```diff
@@ -1214,24 +1184,22 @@ private void checkMatchedFieldsCase(boolean requireFieldMatch, String type) thro
             .fragmentSize(25)
             .highlighterType(type)
             .requireFieldMatch(requireFieldMatch);
-        assertResponse(req.setQuery(twoFieldsQuery).highlighter(new HighlightBuilder().field(fooField).field(barField)), response -> {
+        assertResponses(response -> {
             assertHighlight(response, 0, "foo", 0, equalTo(expectedHighlight1));
             assertHighlight(response, 0, "bar", 0, equalTo("<em>cat</em> <em>cat</em> junk junk junk junk"));
-        });
+        },
+            req.setQuery(twoFieldsQuery).highlighter(new HighlightBuilder().field(fooField).field(barField)),
+            // Setting a matchedField that isn't searched/doesn't exist is simply ignored.
+            req.setQuery(twoFieldsQuery).highlighter(new HighlightBuilder().field(fooField).field(barField.matchedFields("bar", "candy")))
```

Comment:
Same as above; reusing the same `req` instance?

Reply:
Done!